packages/cactus-plugin-bungee-hermes/src/main/typescript/strategy/strategy-ethereum.ts

```diff
@@ -234,22 +218,157 @@ export class StrategyEthereum implements ObtainLedgerStrategy {
+        throw new InternalServerError(`${fn} response is falsy`);
+      }
+      if (!response.callOutput) {
+        throw new InternalServerError(`${fn} response.callOutput is falsy`);
+      }
+      const { callOutput } = response;
+      if (!Array.isArray(callOutput)) {
+        throw new InternalServerError(`${fn} callOutput not an array`);
+      }
+      const allItemsAreStrings = callOutput.every((x) => typeof x === "string");
+      if (!allItemsAreStrings) {
+        throw new InternalServerError(`${fn} callOutput has non-string items`);
+      }
+      return response.callOutput as string[];
+    } else if (connectorOrApiClient instanceof EthereumApi) {
+      const api: EthereumApi = connectorOrApiClient;
+      const response = await api.invokeContractV1(parameters);
+      if (!response) {
+        throw new InternalServerError(`${fn} response is falsy`);
+      }
+      if (!response.status) {
+        throw new InternalServerError(`${fn} response.status is falsy`);
+      }
+      const { status, data, statusText, config } = response;
+      if (response.status < 200 || response.status > 300) {
+        // We log the error here on the debug level so that later on we can inspect the contents
+        // of it in the logs if we need to. The reason that this is important is because we do not
+        // want to dump the full response onto our own error response that is going back to the caller
+        // due to that potentially being a security issue that we are exposing internal data via the
+        // error responses.
+        // With that said, we still need to make sure that we can determine the root cause of any
+        // issues after the fact and therefore we must save the error response details somewhere (the logs)
+        this.log.debug(
+          "EthereumApi non-2xx HTTP response:",
+          data,
+          status,
+          config,
+        );
+
+        // For the caller/client we just send back a generic error admitting that we somehow messed up:
+        const errorMessage = `${fn} EthereumApi error status: ${status}: ${statusText}`;
+        throw new InternalServerError(errorMessage);
+      }
+      if (!data) {
+        throw new InternalServerError(`${fn} response.data is falsy`);
+      }
+      if (!data.callOutput) {
+        throw new InternalServerError(`${fn} data.callOutput is falsy`);
+      }
+      const { callOutput } = data;
+      if (!Array.isArray(callOutput)) {
+        throw new InternalServerError(`${fn} callOutput not an array`);
+      }
+      const allItemsAreStrings = callOutput.every((x) => typeof x === "string");
+      if (!allItemsAreStrings) {
+        throw new InternalServerError(`${fn} callOutput has non-string items`);
+      }
+      return response.data.callOutput;
+    }
+    throw new InternalServerError(
+      `${fn}: neither EthereumApi nor Connector given`,
+    );
+  }
+
+  async invokeWeb3EthMethod(
```

Comment:
@LordKubaya it seems like this method could be inside the ledger connector, instead of being in our package.  Would that make sense/be possible?

Reply:
I don't think that is the case as this is just a function to shadow the calling of the method invokeRawWeb3EthMethod by the API or by the connector, it does not add new funcionalities.